nima_core/embeddings.py

```
@@ -0,0 +1,197 @@
+        embed("ping")
+        return True
+    except RuntimeError:
+        return False
+
+
+# ── Cosine similarity (utility, no external deps) ────────────────────────────
+
+
+def cosine_similarity(a: list[float], b: list[float]) -> float:
+    """Compute cosine similarity between two float vectors."""
+    dot = sum(x * y for x, y in zip(a, b))
```

Ruff (0.15.7)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)